sdk/nodejs/resource.ts

```diff
@@ -174,6 +233,27 @@ export abstract class Resource {
         }
         this.__protect = !!opts.protect;
 
+        // Collapse any `Alias`es down to URNs. We have to wait until this point to do so because we do not know the
```

Comment:
given the complexity of the above and this, consider extracting to function to help keep things separated, and to make it clear waht data is being returned outward.

Reply:
Done.